Setup CSS bundling, add bootstrap/normalize stylesheets

Arthelon opened the pull request:
This PR addresses #1 

- Sets up CSS bundling using Svelte preprocessor hooks
- Adds stylesheet for bootstrap-grid and normalize.css

Arthelon committed:
Add normalize.css to HTML using CDN link

Arthelon committed:
Add bootstrap css CDN link

Signed-off-by: Arthelon <hsing.daniel@gmail.com>

kiootic commented on src/template.html:
```diff
@@ -7,6 +7,8 @@
 
 	%sapper.base%
 
+	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css" integrity="sha256-l85OmPOjvil/SOvVt3HnSSjzF1TUMyT9eV0c2BzEGzU=" crossorigin="anonymous" />
```
I'm not familiar with Svelte/Sapper. Do you know whether it supports bundling CSS stylesheets? We'd like to bundle these stylesheets (as well as our custom styles) directly as one CSS file, so that these external dependency is not needed.

Arthelon replied:
Yes it does support bundling CSS files. I will make this change immediately.

louischan-oursky replied:
It is OK to reference external third CSS files. I am not sure how are we going to use the grid CSS framework. If the grid CSS framework is written in SCSS mixin, then we are going to bundle our CSS. If the grid CSS framework is just a CSS stylesheet, it is fine to reference it with `<link>`. Anyway, we need to bundle SCSS/CSS.

Arthelon replied:
Turns out the styles for bootstrap grid system has its own CSS file. I've bundled it with normalize.css in my latest commit.

kiootic reviewed (commented):
Let's hold this PR after #5 is merged, so that we can test the full CI flow.

Arthelon committed:
Remove CDN links, bundle CSS using PostCSS svelte preprocessor and CSS
imports

Signed-off-by: Arthelon <hsing.daniel@gmail.com>

kiootic commented:
@Arthelon Please rebase on master branch.

Arthelon committed:
Add supported browserslist and autoprefixer PostCSS plugin

Signed-off-by: Arthelon <hsing.daniel@gmail.com>

kiootic commented on src/routes/_layout.svelte:
```diff
@@ -1,5 +1,6 @@
 <script>
   import Nav from "../components/Nav.svelte";
+  import GlobalCSS from "../components/GlobalCSS.svelte"; //eslint-disable-line
```
```suggestion
  import "../components/GlobalCSS.svelte";
```

By the way, if we want to disable some lint rule, we should specify which specific lint rule we are disabling like ` // eslint-disable-line no-use-before-define` and document the reason so that maintainer can know why it is disabled.

kiootic reviewed (commented):
I'm getting this error when running `npm run dev`:
```
✗ client
'import' and 'export' may only appear at the top level
11: }
12: 
13: export function connect(port) {
    ^
14:   if (source || !window.EventSource) return;
```

Do you know why/how to fix it?

louischan-oursky commented on rollup.config.js:
```diff
@@ -1,100 +1,124 @@
+            [
+              "@babel/preset-env",
+              {
+                targets: "> 0.25%, not dead",
```
Since we now have browserlist, [let's not set `targets`](https://babeljs.io/docs/en/babel-preset-env#browserslist-integration)

Arthelon commented:
Seems to be an unresolved issue with rollup/sapper: https://github.com/sveltejs/sapper/issues/1151 . I will add the fixed npm commands for the time being.

Arthelon committed:
Remove @babel/preset-env browser targets, disable specific eslint rule for GlobalCSS import

Signed-off-by: Arthelon <hsing.daniel@gmail.com>

Arthelon committed:
Downgrade @rollup/plugin-commonjs to fix sapper dev bug

Previously, when running the 'dev' npm command. Sapper would spit out
the following error message:
"'import' and 'export' may only appear at the top level "

This issue is related to sapper and the commmonjs rollup plugin: https://github.com/rollup/plugins/issues/304

Resolve temporarily by downgrading @rollup/plugin-commonjs to 11.0.2
Signed-off-by: Arthelon <hsing.daniel@gmail.com>

kiootic commented on src/routes/_layout.svelte:
```diff
@@ -1,5 +1,7 @@
 <script>
   import Nav from "../components/Nav.svelte";
+  // Following error is disabled sinec we are importing GlobalCSS component solely to include global styles
+  import GlobalCSS from "../components/GlobalCSS.svelte"; // eslint-disable-line no-unused-vars
```
I suppose we can just do this?
`import "../components/GlobalCSS.svelte";`

Arthelon committed:
Import GlobalCSS for side-effects only

Signed-off-by: Arthelon <hsing.daniel@gmail.com>
